superdesk/data_updates/00022_20200309-121947_archive.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8; -*-
+# This file is part of Superdesk.
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+#
+# Author  : GyanP
+# Creation: 2020-03-09 12:19
+
+from bson import ObjectId
+from bson.errors import InvalidId
+from superdesk import get_resource_service
+from superdesk.vocabularies import is_related_content
+from superdesk.commands.data_updates import DataUpdate
+
+
+class DataUpdate(DataUpdate):
+
+    resource = 'archive'
+
+    def forwards(self, mongodb_collection, mongodb_database):
+        related = list(get_resource_service('vocabularies').get(req=None, lookup={'field_type': 'related_content'}))
+        archive_service = get_resource_service('archive')
+        for resource in ('archive', 'published'):
+            service = get_resource_service(resource)
+            for item in mongodb_database[resource].find({'associations': {'$gt': {}}}):
```

Comment:
IMO this filter condition is weird. 
Wouldn't it be more explicit to use something like that
`{ associations: { $exists: true, $ne: {} }`  (not tested, just guessed)